src/subtitles/extractor.rs

```diff
@@ -0,0 +1,188 @@
+            range_group.last().unwrap().offset + range_group.last().unwrap().size as u64;
+        let total_size = end_offset - start_offset;
+
+        // Download the chunk
+        stream.seek(SeekFrom::Start(start_offset)).await?;
+        let mut chunk_data = vec![0u8; total_size as usize];
+        stream.read(&mut chunk_data).await?;
+
+        // Extract subtitle entries from this chunk
+        for range in range_group {
+            let relative_offset = (range.offset - start_offset) as usize;
+            let sample_end = relative_offset + range.size as usize;
+
+            if sample_end <= chunk_data.len() {
+                let sample_data = &chunk_data[relative_offset..sample_end];
+
+                // Parse subtitle data based on codec type
+                if let Ok(entries) =
+                    parse_subtitle_sample_data(sample_data, range.timestamp, &track.codec_type)
+                {
+                    subtitle_entries.extend(entries);
+                }
+            }
+        }
+    }
+
+    // Sort by timestamp string (SRT format timestamps sort lexicographically)
+    subtitle_entries.sort_by(|a, b| a.start.cmp(&b.start));
+
+    Ok(subtitle_entries)
+}
+
+/// Calculate optimized subtitle sample ranges
+fn calculate_optimized_subtitle_ranges(
+    track: &SubtitleTrackInfo,
+) -> MediaParserResult<Vec<SubtitleSampleRange>> {
+    let mut ranges = Vec::new();
+    let mut sample_index = 0;
+
+    // Calculate timestamps for samples using timing information
+    //let sample_timestamps = calculate_sample_timestamps(track);
+    let sample_timestamps = build_sample_timestamps(track.timescale, &track.stts_entries);
+
+    // Map samples to chunks and calculate byte ranges
+    for (chunk_idx, &chunk_offset) in track.chunk_offsets.iter().enumerate() {
+        let chunk_num = (chunk_idx + 1) as u32;
+
+        // Find how many samples are in this chunk
+        let samples_in_chunk = get_samples_in_chunk(chunk_num, &track.sample_to_chunk);
+
+        let mut chunk_byte_offset = 0u64;
+        for _ in 0..samples_in_chunk {
+            if sample_index < track.sample_sizes.len() {
+                let sample_size = track.sample_sizes[sample_index];
+
+                // Skip empty samples
+                if sample_size > 0 {
+                    let sample_offset = chunk_offset + chunk_byte_offset;
+                    let timestamp = sample_timestamps.get(sample_index).copied().unwrap_or(0.0);
+
+                    ranges.push(SubtitleSampleRange {
+                        offset: sample_offset,
+                        size: sample_size,
+                        _sample_index: sample_index as u32,
+                        timestamp,
+                    });
+                }
+
+                chunk_byte_offset += sample_size as u64;
+                sample_index += 1;
+            }
+        }
+    }
+
+    Ok(ranges)
+}
+
+/// Calculate sample timestamps from timing tables
+#[allow(dead_code)]
```

Comment:
Why? We don't generally allow unused or commented-out code.

Reply:
Agree. Any dead code should be removed.